src/server/run_sched.c

```diff
@@ -490,57 +398,35 @@ schedule_jobs(pbs_sched *psched)
  *							a socket is closed.
  * @return	void
  */
-
-static void
+void
 scheduler_close(int sock)
 {
-	struct deferred_request *pdefr;
 	pbs_sched *psched;
+	int other_conn;
 
 	psched = find_sched_from_sock(sock);
 
 	if (psched == NULL)
 		return;
 
-	set_attr_generic(&(psched->sch_attr[(int) SCHED_ATR_sched_state]), &sched_attr_def[(int) SCHED_ATR_sched_state], SC_IDLE, NULL, SET);
-
-	if ((sock != -1) && (sock == psched->scheduler_sock2)) {
-		psched->scheduler_sock2 = -1;
-		return;	/* nothing to check if scheduler_sock2 */
-	}
-
-	set_sched_sock(-1, psched);
+	set_sched_state(psched, SC_DOWN);
+	log_eventf(PBSEVENT_SYSTEM, PBS_EVENTCLASS_SCHED, LOG_CRIT, psched->sc_name, "scheduler disconnected");
+	if (sock == psched->sc_primary_conn)
+		other_conn = psched->sc_secondary_conn;
+	else
+		other_conn = psched->sc_primary_conn;
+	psched->sc_secondary_conn = -1;
+	psched->sc_primary_conn = -1;
```

Comment:
other_conn can be either primary or secondary, but why are you assigning -1 to both of them in one shot. scheduler_close can actually get called twice once for primary and once per secondary right ?

Reply:
Yes, if any of the connections is closed, server will close both connection, and if we depend on close_func() of other conn then it will be try to close each other in loop, also see blow we are removing close_func and closing other connection here only.

Reply:
If we are calling close() for both the sockets then it should fine to put -1 for both of them. If by any chance we do not call close() but we are assign -1 to one of the sockets might cause a problem. Please cross check.